spec/workers/decline_offers_by_default_worker_spec.rb

```diff
@@ -0,0 +1,52 @@
+require 'rails_helper'
+
+RSpec.describe DeclineOffersByDefaultWorker do
+  let(:choices) { 3.times.map { create(:application_choice, status: 'offer') } }
+  let(:query_service)   { GetApplicationChoicesReadyToDeclineByDefault }
+  let(:decline_service) { DeclineOfferByDefault }
+
+  before { allow(query_service).to receive(:call).and_return(choices) }
+
+  def invoke_worker
+    DeclineOffersByDefaultWorker.new.perform
+  end
+
+  describe 'processes all application_choices' do
+    it 'processes all application choices returned by the query service' do
+      decline_instance = instance_double(decline_service)
+      allow(decline_instance).to receive(:call)
+      allow(decline_service).to receive(:new).and_return(decline_instance)
+      invoke_worker
+      choices.each do |choice|
+        expect(decline_service).to have_received(:new).with(application_choice: choice)
+      end
+      expect(decline_instance).to have_received(:call).exactly(3).times
+    end
```

Comment:
I'm not sure this example is needed, the functionality is covered by the following one without the need to stub all the dependencies.

Reply:
Yes, will remove